lightning/src/routing/scoring.rs

```diff
@@ -316,6 +316,68 @@ impl ReadableArgs<u64> for FixedPenaltyScorer {
 	}
 }
 
+/// [`Score`] implementation that factors in in-flight HTLC liquidity.
+///
+/// Useful for custom [`Router`] implementations to wrap their [`Score`] on-the-fly when calling
+/// [`find_route`].
+///
+/// [`Router`]: crate::routing::router::Router
+/// [`find_route`]: crate::routing::router::find_route
+pub struct ScorerAccountingForInFlightHtlcs<'a, S: Score>
```

Comment:
Not to beat a dead horse from the offline discussions, but anyone wanting to augment the behavior of `DefaultRouter` (e.g., filter first hops) could always just wrap it and delegate. Then we wouldn't need to expose `ScorerAccountingForInFlightHtlcs` at all. The standalone `find_route` function doesn't really need to be public either now that `Router` is in the `lightning` crate.

Reply:
I'd definitely prefer a more graph-oriented interface as you described. Though, as it stands, `Router`'s interface already takes `InFlightHtlcs`. In either world, we could have a configuration in `DefaultRouter` as to whether to consider in-flight HTLCs. Just seems like `ScorerAccountingForInFlightHtlcs` is an implementation details that is unnecessary to expose.

Reply:
So long as it is well documented that it is for building a `Router`, I would't be entirely opposed. But I'd prefer it live with `Router` to avoid confusion.

Reply:
> We've had folks ask about using the LDK router with CLN/eclair to A/B test routing strategies. You'd probably want to use a `ScorerAccountingForInFlightHtlcs` for that as well.

Not sure I understand how that would be any different from using `DefaultRouter`, which uses `ScorerAccountingForInFlightHtlcs` internally.

Reply:
Isn't "using the LDK router with CLN/eclair" simply reusing `DefaultRouter`? Or is it something else? That's the part that I don't understand.

Reply:
SGTM